driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -186,11 +203,14 @@ private int getOpMsgFlagBits() {
     }
 
     private int getOpMsgResponseExpectedFlagBit() {
-        if (requireOpMsgResponse()) {
-            return 0;
-        } else {
-            return 1 << 1;
+        int flagBits = 0;
+        if (!requireOpMsgResponse()) {
+            flagBits = 1 << 1;
+        }
+        if (exhaustAllowed) {
```

Comment:
This is the new bit: we now have to support the exhaustAllowed bit for streamable isMaster